### PR TITLE
Fix to hide view switch in topology when no project selected

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**Fixes**
https://issues.redhat.com/browse/ODC-4222
https://issues.redhat.com/browse/ODC-4507

**Description**
Fix to not show the grid/list icon when no project is selected or when no items exist in the project (showing Add page). Also updates the Add page background when the user is in list view mode.

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

/kind bug

### CODE REVIEW

File: frontend/packages/dev-console/src/components/topology/TopologyPage.tsx [hunk 1]
```diff
@@ -21,8 +21,11 @@ import TopologyShortcuts from './TopologyShortcuts';
 import { ConnectedTopologyView } from './TopologyView';
 import { LAST_TOPOLOGY_VIEW_LOCAL_STORAGE_KEY } from './components/const';
 import { TOPOLOGY_SEARCH_FILTER_KEY } from './filters';
+import DataModelProvider from './data-transforms/DataModelProvider';
 
 import './TopologyPage.scss';
+import ModelContext, { ExtensibleModel } from './data-transforms/ModelContext';
+import { observer } from '@patternfly/react-topology';
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/topology/TopologyPage.tsx:28]: nit: sort import order

**Comment by karthikjeeyar** [comment]
It looks like there are some test failures

**Comment by jeff-phillips-18** [comment]
@karthikjeeyar Fixed the test failures

File: frontend/packages/dev-console/src/components/topology/TopologyPage.tsx [hunk 2]
```diff
@@ -72,10 +75,16 @@ export function renderTopology({
...
+export const TopologyPageContext: React.FC<TopologyPageProps> = observer(({ match }) => {
   const queryParams = useQueryParams();
   const searchParams = queryParams.get('searchQuery');
   const namespace = match.params.name;
+  const dataModelContext = React.useContext<ExtensibleModel>(ModelContext);
+  const emptyModel =
+    dataModelContext?.isEmptyModel ??
+    (match.params.testEmptyModelValue === undefined
+      ? true
+      : match.params.testEmptyModelValue === 'true');
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/topology/TopologyPage.tsx:87]: @jeff-phillips-18  is it needed only for testing purpose? if yes,  then can we mock the useContext to return true/false in the test cases, WDYT?
eg:
 `spyOn(React, 'useContext').and.returnValue({ isEmptyModel: true });`

File: frontend/packages/dev-console/src/components/topology/__tests__/TopologyPage.spec.tsx [hunk 3]
```diff
@@ -111,18 +112,40 @@ describe('Topology page tests', () => {
...
+      },
+    };
+
+    renderTopologyProps = {
+      model: topologyData,
+      showGraphView: true,
+      loaded: true,
+      loadError: '',
+      namespace: 'topology-test',
+    };
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/topology/__tests__/TopologyPage.spec.tsx:143]: We dont need this as well

**Comment by invincibleJai** [comment]
/lgtm
/approve

Verified changes, works as expected

**Pull request merged**